application/src/main/java/org/thingsboard/server/service/entitiy/ruleChain/DefaultTbRuleChainNotifyService.java

```diff
@@ -0,0 +1,266 @@
+
+    @Override
+    public void delete(RuleChain ruleChain, SecurityUser user) throws ThingsboardException {
+        TenantId tenantId = ruleChain.getTenantId();
+        RuleChainId ruleChainId = ruleChain.getId();
+        try {
+
+
+            List<RuleNode> referencingRuleNodes = ruleChainService.getReferencingRuleChainNodes(tenantId, ruleChainId);
+
+            Set<RuleChainId> referencingRuleChainIds = referencingRuleNodes.stream().map(RuleNode::getRuleChainId).collect(Collectors.toSet());
+
+            List<EdgeId> relatedEdgeIds = null;
+            if (RuleChainType.EDGE.equals(ruleChain.getType())) {
+                relatedEdgeIds = findRelatedEdgeIds(tenantId, ruleChainId);
+            }
+
+            ruleChainService.deleteRuleChainById(tenantId, ruleChainId);
+
+            referencingRuleChainIds.remove(ruleChain.getId());
+
+            if (RuleChainType.CORE.equals(ruleChain.getType())) {
+                referencingRuleChainIds.forEach(referencingRuleChainId ->
+                        tbClusterService.broadcastEntityStateChangeEvent(tenantId, referencingRuleChainId, ComponentLifecycleEvent.UPDATED));
+
+                tbClusterService.broadcastEntityStateChangeEvent(tenantId, ruleChain.getId(), ComponentLifecycleEvent.DELETED);
+            }
+
+            notificationEntityService.notifyDeleteRuleChane(tenantId, ruleChain, relatedEdgeIds, user);
+        } catch (Exception e) {
+            notificationEntityService.notifyCreateOrUpdateOrDelete(tenantId, null, emptyId(EntityType.RULE_CHAIN),
+                null, user, ActionType.DELETED, false, e, ruleChainId.toString());
+            throw handleException(e);
+        }
+    }
+
+    @Override
+    public RuleChain saveRuleChain(TenantId tenantId, DefaultRuleChainCreateRequest request, SecurityUser user) throws ThingsboardException {
+        try {
+            RuleChain savedRuleChain = installScripts.createDefaultRuleChain(tenantId, request.getName());
+            tbClusterService.broadcastEntityStateChangeEvent(tenantId, savedRuleChain.getId(), ComponentLifecycleEvent.CREATED);
+            notificationEntityService.notifyCreateOrUpdateOrDelete(tenantId, null, savedRuleChain.getId(),
+                    savedRuleChain, user, ActionType.ADDED, false, null);
+            return savedRuleChain;
+        } catch (Exception e) {
+            RuleChain ruleChain = new RuleChain();
+            ruleChain.setName(request.getName());
+            notificationEntityService.notifyCreateOrUpdateOrDelete(tenantId, null, emptyId(EntityType.RULE_CHAIN),
+                    ruleChain, user, ActionType.ADDED, false, e);
+            throw handleException(e);
+        }
+    }
+
+    @Override
+    public RuleChain setRootRuleChain(TenantId tenantId, RuleChain ruleChain, SecurityUser user) throws ThingsboardException {
+        RuleChain previousRootRuleChain = ruleChainService.getRootTenantRuleChain(tenantId);
+        RuleChainId ruleChainId = previousRootRuleChain.getId();
```

Comment:
NullPointerException is thrown in case **previousRootRuleChain** is null.

Reply:
According to previous logic, **ruleChainId** should be taken from **ruleChain** variable.

Reply:
Commit https://github.com/thingsboard/thingsboard/pull/6602/commits/a778305b03f8cfbf70ce1262b156d3fd78dece49